### ISSUE TITLE
create a build step to check if a branch is out of date

### ISSUE THREAD

**Issue by chrisdholt** [microsoft/fast-dna#156]
Currently github offers an integrated solution to check if a feature branch is up to date with the latest master branch. While this feature is helpful, there is an action displayed which allows the feature branch to merge master into itself to "bring it up to date". This can create issues and per a response from github there is no plans to change or enhance this experience.

The ideal scenario is to create a build step that is run when updates are pushed to a feature branch which mimics the behavior natively offered by github. If the branch is up to date, the step will pass. If it is not up to date, it will fail. Failure will illustrate that a rebase needs to take place locally to bring the branch up to date.

Github provided the following details on how we can compare two branches:
-------
If you compare two branches using this endpoint:

https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fdeveloper.github.com%2Fv3%2Frepos%2Fcommits%2F%23compare-two-commits&data=02%7C01%7CChristopher.Holt%40microsoft.com%7C0a749ffc269e4c5a656508d595352493%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636578977839271053&sdata=Km8aTFLUokJ%2BTT0813aESM8Gy6QuXOEw82Kh4erodTA%3D&reserved=0

then you'll receive these fields in the response (among other things):

  "status": "behind",
  "ahead_by": 1,
  "behind_by": 2,

That should tell you how the head branch of the compare relates to the base branch of the compare.

**Comment by chrisdholt** [comment]
We should ensure that we consider the various status' that are passed by the github api: https://stackoverflow.com/questions/23943855/github-api-to-compare-commits-response-status-is-diverged

**Comment by chrisdholt** [comment]
I have this mostly complete, however we cannot implement and use this to it's full potential unless builds can automatically be re-triggered when a PR is merged into master.

**Comment by chrisdholt** [comment]
Blocked by limitations to CircleCI and the inability to automatically re-trigger or invalidate builds when something goes into master

**Comment by chrisdholt** [comment]
I'm closing this as obsolete - it no longer applies to our process
